### DISCUSSION TITLE
Good way to understand all the code?

### DISCUSSION THREAD

**Discussion by kragil** [mbrlabs/Lorien#218] [Q&A] [answered]
Hello,

I am interested in understanding all the parts of this project. I tried to start with the latest branch, but at first glance it was a little big. Would it make sense to start with 0.1 and work myself up to 0.6 from there, or has the code been changed a lot, and it wouldn't be a good use of my time?

Thanks

PS: Great project btw

**Comment by mbrlabs** [comment, marked as answer]
Hey, sorry for the late reply. I would start with the latest version i.e. master branch. It's not that much code. 

The project loosely follows a the MVC pattern. You have your UI in the UI folder, your models in Types.gd and in the BrushStroke folder and logic in BrushStroke and InfiniteCanvas. Main.gd ties everything together using Godot's signal system.

When i start out with a project of a certain size i always do it bottom up like this: 
1. Use the program
2. Pick a feature; i.e. the settings dialog
3. Find the UI in the code (that would be UI/SettingsDialog)
4. Change some things and see how it affects the program
5. How does the UI relate to all the other components?
6. Then fix bugs you encounter or see on Github etc ;)

**Reply by kragil** [reply]
Thanks a lot. That is very helpful information.
